include/deal.II/lac/block_linear_operator.h

```diff
@@ -26,23 +26,185 @@
+BlockLinearOperator<n, n, Domain, Range>
+block_back_substitution(const BlockLinearOperator<n, n, Range, Domain> &,
+                        const BlockLinearOperator<n, n, Domain, Range> &);
+
+// end of workaround
+
 /**
- * @name Creation of LinearOperator block structures
+ *  TODO: Description
+ */
+template <size_t m, size_t n, typename Range, typename Domain>
+class BlockLinearOperator : public LinearOperator<Range, Domain>
+{
+public:
+
+  typedef LinearOperator<typename Range::BlockType, typename Domain::BlockType> BlockType;
+  typedef std::array<std::array<BlockType, n>, m> ArrayType;
+
+  /**
+   * Create an empty BlockLinearOperator object. All
+   * <code>std::function</code> member objects are initialized with default
+   * variants that throw an exception upon invocation.
+   */
+  BlockLinearOperator()
+  {
+  }
+
+  /**
+   * Default copy constructor.
+   */
+  BlockLinearOperator(const BlockLinearOperator<m, n, Range, Domain> &) =
+    default;
+
+  /**
+   * Templated copy constructor that creates a BlockLinearOperator object
+   * from an object @p op for which the conversion function
+   * <code>block_operator</code> is defined.
+   */
+  template<typename Op>
+  BlockLinearOperator(const Op &op)
+  {
+    *this = block_operator<m, n, Range, Domain, Op>(op);
+  }
+
+  /**
+   * Default copy assignment operator.
+   */
+  BlockLinearOperator<m, n, Range, Domain> &
+  operator=(const BlockLinearOperator<m, n, Range, Domain> &) = default;
+
+  /**
+   * Templated copy assignment operator for an object @p op for which the
+   * conversion function <code>block_operator</code> is defined.
+   */
+  template <typename Op>
+  BlockLinearOperator<m, n, Range, Domain> &operator=(const Op &op)
+  {
+    *this = block_operator<m, n, Range, Domain, Op>(op);
+    return *this;
+  }
+
+  /**
+   * TODO: A table of the underlying linear operator blocks
+   */
+  ArrayType ops;
+  //@}
+};
+
+
+/**
+ * @name Creation of a BlockLinearOperator
  */
 //@{
 
 /**
  * @relates LinearOperator
  *
- * A function that encapsulates a given collection @p ops of LinearOperators
- * into a block structure. Hereby, it is assumed that Range and Domain are
- * blockvectors, i.e., derived from
- * @ref BlockVectorBase.
- * The individual linear operators in @p ops must act on a the underlying
- * vector type of the block vectors, i.e., on Domain::BlockType yielding a
- * result in Range::BlockType.
+ * A function that encapsulates a @p block_matrix into a
+ * BlockLinearOperator.
+ *
+ * All changes made on individual blocks of @p block_matrix after the
+ * creation of the BlockLinearOperator object are reflected by the operator
+ * object. The block structure of @p block_matrix, however, must not be
+ * changed.
+ *
+ * @ingroup LAOperators
+ */
+template <size_t m, size_t n,
+          typename Range,
+          typename Domain,
+          typename BlockMatrix>
+BlockLinearOperator<m, n, Range, Domain>
+block_operator(const BlockMatrix &block_matrix)
```

Comment:
You could just adopt the usual style for member functions (and functions that are declared in a .h file and implemented in a .cc file) of attaching the documentation with the first declaration, rather than with the implementation.